networks/hamiltonian_net.py

```diff
@@ -108,12 +107,15 @@ def forward(self, q, p):
         x = torch.cat(
             (q, p),
             dim=1)  # Concatenate q and p to obtain a N x 2C x H x W tensor
+        if isinstance(self.paddings[0], list):
```

Comment:
Why is the first one a special case?

Reply:
Is it because its the input layer not defined by the user?

Reply:
Because `paddings` is a list that can contain a `int` value, in which case the padding is intended symmetric, or a list that specifies the padding in all 4 sides. In the first case we already take care of it in the constructor, passing the int value to the `Conv2d` constructor. If the padding is a list instead, we need to create a padding layer in the `forward()`